Gradual and filter task

Albert77708 opened the pull request:
Смог жи?

Albert77708 committed:
commit :W

Albert77708 committed:
adding Action Types

Albert77708 committed:
created filter element without logic

Albert77708 committed:
Added some logic

Albert77708 committed:
added DatePeaker and som logic about

Albert77708 committed:
add part of filter by carriers

Albert77708 committed:
before merge

Albert77708 committed:
Add logic but need to debug

Albert77708 committed:
corrected ActionTypes

Albert77708 committed:
nothing usefull at all

Albert77708 committed:
added loading on scroll

Albert77708 committed:
repair filter by carriers

Albert77708 committed:
added new reducer for combining filtering logic

Albert77708 committed:
repair joint filterin

Albert77708 committed:
fixed sorting logic

Albert77708 committed:
All tasks solved

kir73ul commented on src/Types/Types.ts:
```diff
@@ -1,33 +1,45 @@
+	hasErrored: boolean
+	isLoading: boolean
+	items: TicketsType[]
+	filteredTickets: TicketsType[]
```
нейминг не очень, нужно либо tickets & filteredTickets, либо items & filtreredItems

kir73ul commented on src/Types/Types.ts:
```diff
@@ -1,33 +1,45 @@
+}
+export interface InitialSelectedStateType {
+	pickingDate: string | null
+	sortingItem: ParametersOfFilter | string | null
```
тоже странный нейминг. это же дословно "сортирующий элемент" ?

kir73ul replied:
и что за string там может попасть? любой?

kir73ul commented on src/actions.ts:
```diff
@@ -1,101 +1,130 @@
+
+interface SetSortingItemType {
+	type: typeof actionTypes.SET_SORTING_ITEMS
+	selectItems: ParametersOfFilter | string | null
```
здесь уже selectItems, выше sortingItem, так что же это вообще?

kir73ul commented on src/components/tickets-filter/TicketFilter.tsx:
```diff
@@ -0,0 +1,33 @@
+	const handlerClear = () => {
+		dispatch(SetSortingItem(null))
+	}
+	const treeProps = genTreeProps(allTickets)
```
что значит tree Props ???

kir73ul replied:
не надо нейминг брать от названия элементов антд, надо так чтобы было понятно, что это

kir73ul commented on src/components/tickets-filter/TicketFilter.tsx:
```diff
@@ -0,0 +1,33 @@
+		<div className={styles.wrap}>
+			<TreeSelect
+				onClear={() => handlerClear()}
+				onChange={(value: ParametersOfFilter | string) => handleChange(value)}
```
так почему туда может попасть любая строка?

kir73ul commented on src/components/tickets-filter/helpers/dateFormatter.ts:
```diff
@@ -0,0 +1,3 @@
+export const dateFormatter = (date: string) => {
```
тоже нейминг неинформативный, да может и зачем в отдельную функцию выносить такой простой функционал?

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,33 @@
+import { dateFormatter } from './dateFormatter'
+import { findAllCarriers } from './findAllCarriers'
+
+const day = 86400000
```
millisecondsInDay

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,33 @@
+			return tickets.sort((prev, next) => next.price - prev.price)
+		case ParametersOfFilter.date:
+			return tickets.sort(
+				(prev, next) => dateFormatter(prev.segments[0].date) - dateFormatter(next.segments[0].date)
```
получается мы только по первому элементу сегмента можем сортировать? типа по дате отправления? по дате прибытия нельзя?

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,33 @@
+		case ParametersOfFilter.pickDate:
+			return tickets.filter((ticket) =>
+				date
+					? Math.floor(dateFormatter(ticket.segments[0].date) / day) ===
```
вот это как раз в отдельную функцию можно вынести

kir73ul replied:
но тоже, только по первому сегменту?

kir73ul commented on src/components/tickets-filter/helpers/findAllCarriers.tsx:
```diff
@@ -0,0 +1,37 @@
+	disabled?: boolean
+}
+
+export const findAllCarriers = (tickets: TicketsType[]) => {
```
это очень дорогостоящая операция получается, пробегаться по всем билетам. Всё таки известные все перевозчики лучше описать заранее всё же. По сути в реальном приложении будет либо отдельный запрос на список перевозчиков, либо жёстко на фронте зашиты все перевозчики в константе или енаме и расширяемы по факту добавления новых ( скорее всего первое, более универсально). Лучше всё таки вынеси в енам этот список

kir73ul commented on src/components/tickets-filter/helpers/findAllCarriers.tsx:
```diff
@@ -0,0 +1,37 @@
+	const extractCarriers = tickets.map(({ carrier }) => allDifferenceCarriers.add(carrier))
+	return Array.from(allDifferenceCarriers)
+}
+export const genTreeProps = (allTickets: TicketsType[]) => {
```
нейминг. что за gen ?

kir73ul replied:
плюс у тебя эта дорогостоящая операция findAllCarriers будет выполнятся получается при каждом вызове genTreeProps, который будет вызываться на каждый рендер компонента. А это жесть)

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
-    setFilteredItems(filterTickets(items, stopsFilter));
-    window.scrollTo({ top: 0, behavior: "smooth" });
-  }, [priority, stopsFilter, items]);
+	const scrollHandler = (event: any) => {
```
any ?

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
+		) {
+			setPortionOfItems([
+				...portionOfItems,
+				...filteredTickets.slice(portionOfItems.length, portionOfItems.length + 10),
```
`setPortionOfItems((prevItems) => [...prevItems, ......filteredTickets.slice(portionOfItems.length, portionOfItems.length + 10)]`

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
+		return () => {
+			document.removeEventListener('scroll', scrollHandler)
+		}
+	})
```
а чо без депсов? при каждом ререндере новый листенер добавляем?

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
+	useEffect(() => {
+		const filteredByPriority = sortTickets(items, priority)
+		const filteredBySelect = sortingItem
+			? filterTicketsBySelect(sortingItem, filteredByPriority)
```
не очень логика, лучше в саму функцию filterTicketsBySelect добавить поведение, что если !sortingItem пусть возвращает обратно исходный. Хотя вроде бы там и так уже так)

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
+			? filterTicketsBySelect(sortingItem, filteredByPriority)
+			: filteredByPriority
+		const filteredByPickDate = pickingDate
+			? filterTicketsBySelect(ParametersOfFilter.pickDate, filteredBySelect, pickingDate)
```
и здесь так же.
вообще зря наверное засунул все сортировки в одну функцию, если приходится вызывать на каждый случай снова. Надо просто разбить на отдельные) слишком усложнил логику и соответственно это трудно расширяемо

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,82 @@
+			if (!isLoading && !hasErrored && !portionOfItems.length) {
+				setEmptyList(true)
+			} else setEmptyList(false)
+		}, 200)
```
зачем сеттаймаут? без него никак?

kir73ul commented on src/components/tickets-menu/tickets-menu.tsx:
```diff
@@ -1,33 +1,35 @@
+	const priority = useSelector((state: AppStateType) => state.priorityReducer)
+	const sortingItem = useSelector((state: AppStateType) => state.selectReducer.sortingItem)
+	const dispatch = useDispatch()
+	const isSortingValid =
```
что значит isSortingValid ? а если sortingItem === arametersOfFilter.lowerPrice то уже не валид? )
и sortingItem это получается имя выбранного фильтра? нейминг поправь)

Albert77708 committed:
fixed some errors

Albert77708 committed:
almost all errors fixed

Albert77708 committed:
Refactoring styles with styled component

part 2 of styledComp

part 3 of styledComponent

done styling with styled components

Albert77708 committed:
fixed all remarks

kir73ul commented on src/components/ticket/styled.ts:
```diff
@@ -0,0 +1,59 @@
+import styled from 'styled-components'
```
а как у тебя в эту ветку styled попал?

kir73ul commented on src/components/ticket/styled.ts:
```diff
@@ -0,0 +1,59 @@
+	grid-template-columns: repeat(3, 1fr);
+	grid-template-rows: 2fr repeat(4, 1fr);
+
+	@media (max-width: 476px) {
```
смотрю max-width: 476px используется много раз, 
font-weight: 600;
font-size: 14px;
и еще есть вещи что по несколько раз юзаются. Посмотри как в стайледе использовать переменные или функции для вычисления, попробуй)

kir73ul commented on src/components/tickets-filter/DatePicker.tsx:
```diff
@@ -0,0 +1,28 @@
+		<Wrapper top='295px'>
+			<Space>
+				<DatePicker
+					onChange={(moment, date) => {
```
типы для moment и date ?

Albert77708 replied:
типизировал

kir73ul commented on src/components/tickets-filter/DatePicker.tsx:
```diff
@@ -0,0 +1,28 @@
+import { Space, DatePicker } from 'antd'
+import { useDispatch } from 'react-redux'
+import { setPickingDate } from '../../actions'
+import { Wrapper } from './styled'
```
короч у тебя весь styled здесь? в однй ветке чтоли всё делал и два ПРа от одной ветке?

Albert77708 replied:
да забыл создать новую ветку

kir73ul commented on src/components/tickets-filter/TicketFilter.tsx:
```diff
@@ -0,0 +1,34 @@
+	return (
+		<Wrapper>
+			<TreeSelect
+				onClear={() => handlerClear()}
```
странно, в доке анта не нашёл onClear у этого компонента. Он точно работает? )

Albert77708 replied:
Работает видос прикладываю

https://user-images.githubusercontent.com/84145795/151671465-9e9ac20e-d884-4d6b-8d98-d7aa9f7e98e2.mp4

kir73ul commented on src/components/tickets-filter/TicketFilter.tsx:
```diff
@@ -0,0 +1,34 @@
+		<Wrapper>
+			<TreeSelect
+				onClear={() => handlerClear()}
+				onChange={(value: ParametersOfFilter | Carriers) => handleChange(value)}
```
и `onClear={handlerClear}`
`onChange={handleChange}`

Albert77708 replied:
да обе работают

kir73ul commented on src/components/tickets-filter/TicketFilter.tsx:
```diff
@@ -0,0 +1,34 @@
+				treeIcon={true}
+				allowClear={true}
+				treeDefaultExpandAll
+			></TreeSelect>
```
просто />

Albert77708 replied:
исправлено

kir73ul commented on src/components/tickets-filter/helpers/dateSorting.ts:
```diff
@@ -0,0 +1,12 @@
+			? Math.floor(Date.parse(ticket.segments[0].date) / millisecondsInDay) ===
+			  Math.floor(Date.parse(date) / millisecondsInDay)
+			: ticket
+	)
```
вынеси date из маппинга, это ни к чему, оно ж не меняется. `if (!date) return tickets` верно же?

kir73ul replied:
и это тоже вынеси в функцию выше по коду, 
`Math.floor(Date.parse(date) / millisecondsInDay`

Albert77708 replied:
1) вынес
2) в функцию вынес 
![image](https://user-images.githubusercontent.com/84145795/151672135-fe4c7271-75b7-4b73-8270-fbded0a9cd6a.png)

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,30 @@
+	tickets: TicketsType[],
+	date?: string | null
+) => {
+	if (!selectValue) return tickets
```
по факту лишняя строчка, отработает же default case

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,30 @@
+				(prev, next) => Date.parse(prev.segments[1].date) - Date.parse(next.segments[1].date)
+			)
+		case ParametersOfFilter.pickDate:
+			return date ? sortByPickedDate(tickets, date) : tickets
```
что-то дублируется логика здесь и внутри функции. По факту значит в функцию уже null | undefined не попадёт же

kir73ul replied:
либо в функции оставь логику, либо здесь. Лучше же конечно в функции, а здесь убери проверку на date (ну и в функции поправь по комментам)

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -0,0 +1,30 @@
+			)
+		case ParametersOfFilter.pickDate:
+			return date ? sortByPickedDate(tickets, date) : tickets
+		case selectValue as Carriers:
```
а это явный хак. `selectValue as Carriers` ты говоришь, что любое selectValue дошедшее сюда станет иметь тип Carriers, что есть ошибка. Естественно, дефолтный или любые следующие кейсы некогда не выполнятся. Или ты специально сделал этот хак здесь? типа если ни один из фильтров, но что то есть, то это полюбому авиакомпания?
так то норм выход, поскольку есть типизация в самой функции выше и кроме ParametersOfFilter | Carriers
ничего попасть не может.
просто имей ввиду, есть такие нюансы)

Albert77708 replied:
cлучайно получилось не нашел другого способа, может подскажешь как можно сказать типа вот в случае если значение одно из такого то ЕНАМ тогда такой кейс)
Из за этого хака и условие нельзя убрать то что выше прокоментил

kir73ul replied:
Ну есть такая идея: в default кейсе проверять с помощью includes например, является ли одной из авиакомпаний. А если нет, то дефолтное действие. 
Только энам не подойдёт, нужен массив)

kir73ul commented on src/components/tickets-filter/helpers/generateTreeForSelection.tsx:
```diff
@@ -0,0 +1,30 @@
+		{ children: [], value: ParametersOfFilter.airCompany, disabled: true, title: 'Авиакомпании' },
+	]
+	for (const carrier in Carriers) {
+		const iconUrl = url.carrierLogo.replace('carrier', carrier)
```
тут вот опасный момент, при каждом вызове функции буду вновь грузиться картинки. Надо вынести уже созданные один раз иконки из этой функции, здесь только подставлять готовые иконки

Albert77708 replied:
вынести руками ? а в случае если будут меняться авиакомпании?

kir73ul replied:
Нужно само создание иконок отсюда вынести, создать их один раз и уже готовые иконки вставлять

kir73ul replied:
Подойдёт, например, объект для этого. С ключами - именами авиакомпаний, и по второму ключу - иконка

Albert77708 replied:
тут только вот обратил внимание,  что у нас при рендеринге билетов  запрос идет каждый раз на иконки, вот там вот реально может быть чувствительно, а тут два раза функция вызывается за все время, это наверное и не страшно, но переделаю, и тут и там тогда

kir73ul replied:
Кстати есть предложение ещё лучше, скачать все эти иконки один раз и в статику положить в проекте и брать уже оттуда. В таком случае можно и один компонент использовать для иконки, куда имя иконки подставляешь по авиакомпании. Попробуй так сделать, ещё лучше и проще)

kir73ul commented on src/components/tickets-filter/helpers/generateTreeForSelection.tsx:
```diff
@@ -0,0 +1,30 @@
+	disabled?: boolean
+}
+
+export const generateTreeForSelection = (allTickets: TicketsType[]) => {
```
и зачем тебе вообще эта функция? и allTickets же не юзается?

kir73ul replied:
просто treeForSelection инициализируешь и возвращаешь, нафиг тут функция

Albert77708 replied:
просто мысль была динамически закидывать в tree те авиакомпании которые могут добавиться

kir73ul replied:
Ну раз она не оправдалась, то надо убрать лишнее)

kir73ul commented on src/components/tickets-filter/styled.ts:
```diff
@@ -0,0 +1,14 @@
+	position?: string
+	top?: string
+	width?: string
+	marginTop?: string
```
типы тут надо описать лучше. Любые стринги ты же не можешь передать сюда, верно? а по коду написано, что можешь )

Albert77708 replied:
bro, не могу найти как типизировать свойства размерные - топ, width  и тд
не подскажешь как?

kir73ul commented on src/components/tickets-filter/styled.ts:
```diff
@@ -0,0 +1,14 @@
+	marginTop?: string
+}
+export const Wrapper = styled.div<WrapProps>`
+	position: ${(props) => props.position || 'sticky'};
```
я обычно так пишу `${({ position }) => position || 'sticky'};`

kir73ul commented on src/components/tickets-list/error-message.tsx:
```diff
@@ -1,13 +1,11 @@
-  return hasErrored
-    ? (<div className={styles.error} > Sorry! There was an error loading the tickets </div>)
-    : null;
+	return hasErrored ? (
```
можно чуть упростить с исползованием &&

Albert77708 replied:
тут мне кажется не подойдет, т.к. в случае если  hasErrored будет равен фолс,  то и компонент будет возвращать фолс, а это недопустимо, он возвращает либо разметку либо null, или чего то неправильно понимаю?

kir73ul replied:
А как реакт рендерит false?)

Albert77708 replied:
ну по идее не рендерит он фолс, пропускает, но видишь у меня ошибка 
![image](https://user-images.githubusercontent.com/84145795/151690875-f1cf8937-bfe7-4385-b861-c3aa3712cf38.png)
но TS ругается на это

kir73ul replied:
Оберни в <></>

Albert77708 replied:
![image](https://user-images.githubusercontent.com/84145795/151691099-a2ca788d-3ff9-4bc2-a3b7-a3b1c69e9b9b.png)
ну да так работает)

kir73ul commented on src/components/tickets-list/list.tsx:
```diff
@@ -1,20 +1,19 @@
-    const id = idBase.create();
-    return <Ticket key={id} {...props} />;
-  });
+	const elems = ticketPortion.map((item: TicketsType) => {
```
ТS для item сам тип не определяет здесь разве?

Albert77708 replied:
определяет)) убрал

kir73ul commented on src/components/tickets-list/no-elems.tsx:
```diff
@@ -1,7 +1,19 @@
+	const hasErrored = useSelector((state: AppStateType) => state.ticketsReducer.hasErrored)
+	const ticketPortion = useSelector((state: AppStateType) => state.ticketsReducer.ticketPortion)
+	const isLoading = useSelector((state: AppStateType) => state.ticketsReducer.isLoading)
+	const [isEmptyList, setIsEmptyList] = useState(false)
```
вот нафига тебе здесь локальный стейт??? ))) и юзэффект с ним
просто `return !isLoading && !hasErrored && !ticketPortion.length && <Notification>Sorry! No flights</Notification>`

kir73ul replied:
ну или в отдельную переменную вынеси `const IsEmptyList = !isLoading && !hasErrored && !ticketPortion.length`, так лучше читаемо. Но нафига локальный стейт здесь??? )

Albert77708 replied:
ипа затупил то))) ппц )) усложнил все как индус
исправил

kir73ul commented on src/components/tickets-list/tickets-list.tsx:
```diff
@@ -1,39 +1,49 @@
+			document.documentElement.scrollHeight -
+			(window.innerHeight + document.documentElement.scrollTop)
+		if (isOnBottom < 100 && isOnBottom > 1) {
+			setCountOfPortion((prevState) => prevState + 1)
```
вот здесь написать `dispatch(setTicketPortion(filteredTickets))` и не нужен локальный стейт и юзэффект. Или я в чём то не прав?

kir73ul commented on src/components/tickets-menu/styled.ts:
```diff
@@ -0,0 +1,49 @@
+import styled from 'styled-components'
+
+interface Props {
+	isButtonActive: boolean
```
просто isActive

kir73ul replied:
а Props наоборот лучше назвать как ButtonProps

kir73ul commented on src/components/tickets-menu/tickets-menu.tsx:
```diff
@@ -1,33 +1,37 @@
+	const priority = useSelector((state: AppStateType) => state.priorityReducer)
+	const sortingItem = useSelector((state: AppStateType) => state.selectReducer.sortingItem)
+	const dispatch = useDispatch()
+	const isSortingValid =
```
нейминг непонятен) isSortingValid

kir73ul commented on src/components/transfers-filters/styled.ts:
```diff
@@ -0,0 +1,78 @@
+	&::before {
+		content: ' ';
+		border-color: ${(props) => (props.checked ? '#0b76ef' : '')};
+		background: ${(props) => (props.checked ? `center no-repeat url(${image}) ` : '')};
```
пустые строки передавать в цсс свойства цвета не оч правильно

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
 
 const priorityReducer = (state = SortOfTickets.cheapest, action: PriorityReducerType) => {
-  const { type, value } = action;
+	const { type, value } = action
```
переименуй везде value в payload пожалуйста. Я когда делал, хуйню какую то сделал), а общепризнанный нейм для этого, это payload

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
+	}
 }
 
 const initialTicketsState: initialTicketsStateType = {
```
странное название для типа. есть еще и TicketsStateType и они различны с этим initialTicketsStateType ?

Albert77708 replied:
TicketsStateType нее такого нет, а чо нэйминг соответствует названию редьюсера, у каждого редьюсера же свой стэйт  как то его нужно называть )

kir73ul replied:
initial из названия типов нужно убрать. Типы же не для инишл значения, а для стэйта. Да и с маленькой буквы типы не называют

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
+		case actionTypes.ITEMS_HAS_ERRORED:
+			return {
+				...state,
+				hasErrored: action.value,
```
везде пожалуйста переименуй в action.payload

Albert77708 replied:
переименовал, а я же могу называть по другому  не пэйлоад а например как ниже ?
![image](https://user-images.githubusercontent.com/84145795/151691392-8baaac43-b852-404c-b9e7-bf80af70de35.png)

kir73ul replied:
Не, надо  payload и если несколько свойств, то payload тогда становится объектом с этими полями

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
+
+		case actionTypes.SET_TICKET_PORTION:
+			const portionLength = state.ticketPortion.length
+			return action.filteredTickets.length > portionLength && action.filteredTickets.length > 10
```
10 магическое число вынеси в константу осмысленно названную)

kir73ul replied:
а вообще логика странная: разве ты берешь билеты не из одной кучи? думаю можно обойтись вообще без передачи билетов в экшен, а просто по экшену добавлять +10 билетов из отсортированных сейчас. Не вижу никаких проблем, если есть проблемы, то давай обсудим)

kir73ul replied:
и размеры порций эти здесь не сравнивать. Всё гораздо проще мне видится здесь

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
+			return state
+	}
+}
+const initialSelectState: InitialSelectedStateType = {
```
тоже нейминг

kir73ul commented on src/reducer.ts:
```diff
@@ -1,109 +1,150 @@
+		case actionTypes.SET_PICKING_DATE:
+			return {
+				...state,
+				pickingDate: action.date,
```
payload please

kir73ul commented on src/selectors.ts:
```diff
@@ -0,0 +1,25 @@
+export const selectPickingDate = (state: AppStateType) => state.selectReducer.pickingDate
+export const selectPriority = (state: AppStateType) => state.priorityReducer
+export const selectStops = (state: AppStateType) => state.transfersReducer
+export const selectTickets = (state: AppStateType) => state.ticketsReducer.tickets
```
а что у тебя, часть вынесена в селекторы сюда, а выше по коду много было и без селекторов? приведи всё к единому виду, добавь все селекторы сюда

kir73ul reviewed (commented):
35 comments to fix

kir73ul commented on src/components/tickets-filter/DatePicker.tsx:
```diff
@@ -1,26 +1,27 @@
-	const handleClear = () => {
-		dispatch(setPickingDate(null))
+	const handleChange = (moment: moment.Moment | null, date: string) => {
+		moment ? dispatch(setPickingDate(date)) : dispatch(setPickingDate(null))
```
можно ещё проще: `dispatch(setPickingDate(moment ? date : null)`

kir73ul commented on src/components/tickets-filter/DatePicker.tsx:
```diff
@@ -1,26 +1,27 @@
 	}
 	return (
-		<Wrapper top='295px'>
+		<Wrapper top='279.67px'>
```
почему именно такой размер? ставь 280 просто)

kir73ul commented on src/components/tickets-filter/helpers/filterTicketsBySelect.ts:
```diff
@@ -13,15 +20,11 @@ export const filterTicketsBySelect = (
 		case ParametersOfFilter.biggerPrice:
 			return tickets.sort((prev, next) => next.price - prev.price)
```
.sort изменяет исходный массив, имей ввиду

kir73ul commented on src/components/tickets-filter/helpers/treeForSelection.tsx:
```diff
@@ -0,0 +1,51 @@
+			{
+				id: Carriers.EK,
+				value: Carriers.EK,
+				title: <img src={carrierIcons[Carriers.EK]} alt='' width={'80px'} height={'35px'}></img>,
```
одно и то же повторяется много раз, тебя не смущает? )

kir73ul commented on src/components/tickets-filter/styled.ts:
```diff
@@ -1,14 +1,39 @@
-	top: ${(props) => props.top || '245px'};
-	width: ${(props) => props.width || '206.4px'};
-	margin-top: ${(props) => props.marginTop || '20px'};
+	top: ${(props) => props.top || '235.67px'};
```
зачем такие значения? 235.67 ? никто никогда не заметит десятых и сотых пикселя, округляй до целых

kir73ul commented on src/components/tickets-filter/styled.ts:
```diff
@@ -1,14 +1,39 @@
-	width: ${(props) => props.width || '206.4px'};
-	margin-top: ${(props) => props.marginTop || '20px'};
+	top: ${(props) => props.top || '235.67px'};
+	margin: ${(props) => props.marginTop || '12px'} 20px 0 0;
```
корректнее во всех случаях здесь юзать ?? ( ибо условно 0 можно передать, но отработает логически не так )

Albert77708 committed:
fixed new bags and remarks
